src/PKSim.Presentation/Services/CompoundAlternativeTask.cs

```diff
@@ -313,11 +313,12 @@ public TableFormula SolubilityTableForPh(ParameterAlternative solubilityAlternat
          var gainPerCharge = solubilityAlternative.Parameter(CoreConstants.Parameters.SOLUBILITY_GAIN_PER_CHARGE);
          var refSolubilityValue = refSolubility.Value;
 
+         var compoundForCalculation = _executionContext.Clone(compound);
```

Comment:
Creating a clone of the compound before using it to perform calculation. This ensure that we are not MUTATING the compound given as parameter (Template RefPh and Gain where changed, hence the difference when saving loading etc...) Very interesting